src/commands/transaction/mosaicaddressrestriction.ts

```diff
@@ -0,0 +1,109 @@
+import { MosaicIdAliasResolver } from '../../resolvers/mosaic.resolver';
+import { RestrictionKeyResolver } from '../../resolvers/restrictionKey.resolver';
+import { RestrictionValueResolver } from '../../resolvers/restrictionValue.resolver';
+import { AddressAliasValidator } from '../../validators/address.validator';
+import { MosaicIdAliasValidator } from '../../validators/mosaicId.validator';
+import { NumericStringValidator } from '../../validators/numericString.validator';
+import { MosaicRestrictionKeyValidator } from '../../validators/restrictionKey.validator';
+
+export class CommandOptions extends AnnounceTransactionsOptions {
+    @option({
+        flag: 'm',
+        description: 'Mosaic identifier or @alias being restricted.',
+        validator: new MosaicIdAliasValidator(),
+    })
+    mosaicId: string;
+
+    @option({
+        flag: 'a',
+        description: 'Address or @alias being restricted.',
+        validator: new AddressAliasValidator(),
+    })
+    targetAddress: string;
+
+    @option({
+        flag: 'k',
+        description: 'Restriction key.',
+        validator: new MosaicRestrictionKeyValidator(),
+    })
+    restrictionKey: string;
+
+    @option({
+        flag: 'v',
+        description: '(Optional) Previous restriction value.',
+    })
+    previousRestrictionValue?: string;
+
+    @option({
+        flag: 'V',
+        description: 'New restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    newRestrictionValue: string;
+}
+
+@command({
+    description: 'Set a mosaic restriction to an specific address',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+    @metadata
+    execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
+        const account = profile.decrypt(options);
+        const mosaicId = new MosaicIdAliasResolver().resolve(options);
+        const targetAddress = new TargetAddressResolver().resolve(options);
+        const restrictionKey = new RestrictionKeyResolver().resolve(options);
+        const newRestrictionValue = new RestrictionValueResolver().resolve(options);
+        const maxFee = new MaxFeeResolver().resolve(options);
+        let previousRestrictionValue: UInt64 | undefined;
+        if (!options.previousRestrictionValue || '' === options.previousRestrictionValue) {
+            previousRestrictionValue = undefined;
+        } else {
+            previousRestrictionValue = UInt64.fromNumericString(options.previousRestrictionValue);
+        }
+        const mosaicAddressRestrictionTransaction = MosaicAddressRestrictionTransaction.create(
+            Deadline.create(),
+            mosaicId,
+            restrictionKey,
+            targetAddress,
+            newRestrictionValue,
+            profile.networkType,
+            previousRestrictionValue,
+            maxFee,
+        );
+
+        const networkGenerationHash = profile.networkGenerationHash;
```

Comment:
Add support for the recently added ``shouldAnnounce`` and ``sync`` options:

https://github.com/nemtech/nem2-cli/blob/master/src/commands/transaction/accountlink.ts#L69-L75